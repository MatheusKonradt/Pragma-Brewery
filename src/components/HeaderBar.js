import React from 'react';
import PropTypes from 'prop-types';
import './HeaderBar.sass';
import Icon from './Icon';
import AsideMenu from "./AsideMenu";
import Text from "./Text";

class HeaderBar extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      asideMenuOpen: false
    };
    this.classes = {
      root: 'HeaderBar',
      asideMenuOpen: '-open',
      asideMenuClosed: '',
      menu: 'menu',
      headerTitle: 'headerTitle'
    };
    this.themes = {};
  }

  onClickIconMenu() {
    this.setState({ asideMenuOpen: !this.state.asideMenuOpen });
  }

  render() {
    return (
      <header className={[this.classes.root, this.props.className].join(' ')} style={this.props.themes.root}>
        <a href="#" onClick={this.onClickIconMenu.bind(this)} className={this.classes.menu} >
          <Icon name="dehaze"/>
        </a>
        <Text variant="h1" className={this.classes.headerTitle}>Pragma Brewery</Text>
        <AsideMenu open={this.state.asideMenuOpen}/>
      </header>
    );
  }
}

HeaderBar.defaultProps = {
  themes: {},
  className: '',
};

HeaderBar.propTypes = {
  themes: PropTypes.object,
  className: PropTypes.string,
};

export default HeaderBar;
