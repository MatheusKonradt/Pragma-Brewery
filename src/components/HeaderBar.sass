@import '../etc/vars'

.HeaderBar
  position: fixed
  top: 0
  width: 100%
  height: 48px
  border-bottom: 1px solid #000
  display: flex
  justify-content: flex-start
  background: #fccd4d
  background: linear-gradient(to right, #fccd4d 0%,#fbdf93 100%)
  z-index: $zindexMenu

  a.menu
    padding: 13px
    color: #000
    text-decoration: none
    z-index: $zindexMenuButton

  .Text.headerTitle
    text-align: center
    position: absolute
    width: 100%
    margin-left: 10px
    color: #000
    font-size: 1.8em
    line-height: 48px

