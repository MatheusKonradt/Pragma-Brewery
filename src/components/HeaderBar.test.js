import React from 'react';
import HeaderBar from './HeaderBar';
import renderer from 'react-test-renderer';

test('Should match snapshot', () => {
  const component = renderer.create(
    <HeaderBar />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});