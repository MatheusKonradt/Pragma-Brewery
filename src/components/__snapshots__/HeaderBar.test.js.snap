// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot 1`] = `
<header
  className="HeaderBar "
>
  <a
    className="menu"
    href="#"
    onClick={[Function]}
  >
    <i
      className="Icon material-icons  size-24"
    >
      dehaze
    </i>
  </a>
  <h1
    className="Text headerTitle"
  >
    Pragma Brewery
  </h1>
  <div
    className="AsideMenu "
  >
    <div
      className="menu"
      onClick={[Function]}
    />
    <div
      className="overlay"
      onClick={[Function]}
    />
  </div>
</header>
`;
